### DISCUSSION TITLE
Building Daylight Distribution Process

### DISCUSSION THREAD

**Discussion by hadminh** [onthegomap/planetiler#729] [Q&A] [answered]
|                                                                   Input                                                                   | Version |             Machine             |              Time               | mbtiles size |                                                          Logs                                                          |
|-------------------------------------------------------------------------------------------------------------------------------------------|---------|---------------------------------|---------------------------------|--------------|------------------------------------------------------------------------------------------------------------------------|
[Daylight Distribution v1.6](https://daylightmap.org/2021/09/29/daylight-v16-released.html) with ML buildings and admin boundaries (67GB) | 0.1.0   | DO 16cpu 128GB                  | 3h13m cpu:43h40m avg:13.5       | 101GB        | [logs](planet-logs/v0.1.0-daylight-do-16cpu-128gb.txt)                                                                 |

Hi thanks for doing a great work for geo community. I have noticed you have tested the Daylight Distribution.
I am wondering if you have ran the following command to produce the planet file?

```bash
# running apply-changes I notice that i ran out of memory when I include building 
# DO 24cpu 192GB 3xSSD
osmium apply-changes planet-v1.34.osm.pbf admin-v1.34.osc.gz ml-buildings-v1.34.osm.pbf -o everything.osm.pbf
osmium renumber everything.osm.pbf -o planet-daylight.osm.pbf

java -Xmx150g \
  `# return unused heap memory to the OS` \
  -XX:MaxHeapFreeRatio=40 \
  -jar planetiler.jar \
  --area=planet --bounds=planet --osm_path=/home/uesr/planet-daylight.osm.pbf \
  --maxzoom=15 --render_maxzoom=15 \
  `# Accelerate the download by fetching the 10 1GB chunks at a time in parallel` \
  --download-threads=10 --download-chunk-size-mb=1000 \
  `# Also download name translations from wikidata` \
  --fetch-wikidata \
  --output=output.mbtiles \
  `# Store temporary node locations in memory` \
  --nodemap-type=array --storage=ram
```

**Comment by msbarry** [comment, marked as answer]
The process has changed a bit recently as the daylight distribution has gotten bigger. Try these steps: https://gist.github.com/tyrauber/ad03568f399521e9c3b3189aa99e3b51

**Reply by hadminh** [reply]
Thanks for the reply. So Can I assume because we are merging there is no need for renumbering after merging in this [line](https://gist.github.com/tyrauber/ad03568f399521e9c3b3189aa99e3b51#file-daylight-maps-buildscript-L27)

**Reply by tyrauber** [reply]
@hadminh, That is correct. The id conflicts have been resolved upstream. Just apply the .osc sidecars, then merge in the building .pbf after to minimize build memory.

**Reply by hadminh** [reply]
Thank you so much for the script. It is much much much more RAM efficient than what I originally was trying to do!
